### PR TITLE
Revised list patterns reference

### PR DESCRIPTION
**Pull request by pkulikov**
Moved the conceptual-like part to the overview article. Added more examples (matching a string, using a subpattern with `..`)

### CODE REVIEW

File: docs/csharp/language-reference/operators/patterns.md [hunk 1]
```diff
@@ -70,9 +71,9 @@ Beginning with C# 9.0, for that purpose you can use a *type pattern*, as the fol
 
 Like a declaration pattern, a type pattern matches an expression when an expression result is non-null and its run-time type satisfies any of the conditions listed above.
 
-You can also use this pattern for a clear, concise `null` check:
+To check for non-null, you can use a [negated](#logical-patterns) `null` [constant pattern](#constant-pattern), as the following example shows:
```
Comment (pkulikov) [inline docs/csharp/language-reference/operators/patterns.md:74]: nit-picking here: `not null` is neither a declaration nor type pattern

File: docs/csharp/language-reference/operators/snippets/patterns/DeclarationAndTypePatterns.cs [hunk 2]
```diff
@@ -12,17 +12,15 @@ public static void Examples()
...
             if (input is not null)
             {
-                Console.WriteLine(input);
-            } else
-            {
-                throw new ArgumentNullException(paramName: nameof(input), message: "Input should not be null");
+                // ...
             }
-            // </NotNull>
+            // </NonNullCheck>
```
Comment (pkulikov) [inline docs/csharp/language-reference/operators/snippets/patterns/DeclarationAndTypePatterns.cs:21]: For consistency, using the same example for `not null` as in other sections of the patterns reference

Comment (IEvangelist) [reply docs/csharp/language-reference/operators/snippets/patterns/DeclarationAndTypePatterns.cs:21]: I like that.

**Review by IEvangelist — APPROVED** [review]
Looks great, just a few nits.

File: docs/csharp/language-reference/operators/patterns.md (hunk 1 above)
Comment (IEvangelist) [reply docs/csharp/language-reference/operators/patterns.md:74]: You're correct, it is a negated constant pattern.

**Comment by pkulikov** [comment]
@IEvangelist thank you, all suggestions are applied

**Pull request merged**